### PR TITLE
Support closure expr

### PR DESCRIPTION
**Pull request by avitex**
Alternative implementation as discussed in #17

### CODE REVIEW

**Comment by avitex** [comment]
This would allow updating in `core-foundation` like so: https://github.com/avitex/core-foundation-rs/commit/6c146547d74d1e24214989da19bfd51351e4107b

**Comment by avitex** [comment]
I must have been high when I said it is a breaking change. The type changed is pub but within a private module.

File: foreign-types-macros/src/parse.rs [hunk 1]
```diff
@@ -71,6 +77,30 @@ impl Parse for ForeignType {
     }
 }
 
+pub enum ForeignTypeFn {
```
Comment (sfackler) [inline foreign-types-macros/src/parse.rs]: I would just parse an `Expr` and let the compiler complain on its own if it's not callable.

**Comment by sfackler** [comment]
This could definitely use some tests.

**Comment by avitex** [comment]
I've put two cases within test for a closure, a closure with a cast both with `unsafe` usage. If there is something specific you'd like to test for I'm happy to update accordingly.

**Comment by avitex** [comment]
@sfackler Hey mate, just a nudge on this if you get the chance to pull this through. I was going to push a PR with some changes upstream and I'll like to include these changes if you're happy with them :)

**Comment by sfackler** [comment]
Sorry! Somehow filtered filtered this out of my inbox/brain.

**Comment by avitex** [comment]
Thanks a ton :)

**Pull request merged**